### ISSUE TITLE
Is disparity value supposed to represent real world metrics?

### ISSUE THREAD

**Issue by maliksyria** [Totoro97/f2-nerf#101]
Hello! 
Thanks for this very good work! 
I have a question, I'm rendering my own scenes and both of rendering and disparity look very good.
The idea that I'm rendering a stereo pair - One is the original image (as left image), and another one is an image moved on x axis by a virtual baseline value `b` (as right image). 
I'm taking it from this line 
https://github.com/Totoro97/f2-nerf/blob/68755d3f4b7108b0750d4cac1932c2f2dae5c8ac/src/Renderer/Renderer.cpp#L207C1-L207C1
`  Tensor disparity = FlexOps::Sum(weights / sampled_t, idx_start_end);
`
The issue that, when I render the disparity using that baseline 'b' and focal length calculated by COLMAP, I'm getting visually excellent results but odd numbers that does not really represent the expected disparity. My question is F2-NeRF supposed to return realistic disparity values in such a case or not ?

**Comment by maliksyria** [comment]
I'm sorry for tagging you @Totoro97 . Could you please take a look? What do you think this issue is related for? 
Is it related to warping the space or to the perspective sampling strategy?. For all scene I'm having this radial distorted point cloud

**Comment by maliksyria** [comment]
For those who has this problem, during rendering mode, comment out this line https://github.com/Totoro97/f2-nerf/blob/68755d3f4b7108b0750d4cac1932c2f2dae5c8ac/src/PtsSampler/PersSampler.cu#L319
Another normalization on rays' direction before sampling is causing this issue.

**Comment by xubin1994** [comment]
> For those who has this problem, during rendering mode, comment out this line
> 
> https://github.com/Totoro97/f2-nerf/blob/68755d3f4b7108b0750d4cac1932c2f2dae5c8ac/src/PtsSampler/PersSampler.cu#L319
> 
> 
> Another normalization on rays' direction before sampling is causing this issue.

I am also doing something similar, using the ngp_fox data set for testing, and I found that even if I comment out this line of code, the results of using fb/depth (render_result.depth) still cannot match gt. Do you have any corresponding questions? I have a question, is the depth rendered here a distance map instead of a depth map?

**Comment by maliksyria** [comment]
The rendered depth you are taking from left or right image? It must be the left one
Have you tried to see how the nerf_depth = 1/nerf_disp looks like ? @xubin1994

**Comment by xubin1994** [comment]
> The rendered depth you are taking from left or right image? It must be the left one Have you tried to see how the point cloud of nerf_depth = 1/nerf_disp looks like by using Open3D for example? @xubin1994

yes, i used the left one.Why use nerf_depth = 1/nerf_disp? What does this mean? I used gt_disp to convert the point cloud, and the result was correct and without distortion.

**Comment by maliksyria** [comment]
I meant to double check how the depth (not disparity) rendered from F2 after your calculations look like ?
